integrationtests/src/test/java/io/kroxylicious/proxy/ApiVersionsIT.java

```diff
@@ -0,0 +1,123 @@
+            Response response = whenGetApiVersionsFromKroxylicious(client);
+            assertKroxyliciousResponseOffersApiVersionsForApiKey(response, ApiKeys.METADATA, brokerOldestVersion, ApiKeys.METADATA.latestVersion());
+        }
+    }
+
+    @Test
+    public void shouldOfferTheMaximumLowestSupportedVersionWhenKroxyliciousIsAheadOfBroker() {
+        try (var tester = mockKafkaKroxyliciousTester(KroxyliciousConfigUtils::proxy);
+                var client = tester.singleRequestClient()) {
+            short brokerOldestVersion = (short) (ApiKeys.METADATA.oldestVersion() - 1);
+            givenMockRespondsWithApiVersionsForApiKey(tester, ApiKeys.METADATA, brokerOldestVersion, ApiKeys.METADATA.latestVersion());
+            Response response = whenGetApiVersionsFromKroxylicious(client);
+            assertKroxyliciousResponseOffersApiVersionsForApiKey(response, ApiKeys.METADATA, ApiKeys.METADATA.oldestVersion(), ApiKeys.METADATA.latestVersion());
+        }
+    }
+
+    // TODO we think this is a bug but will come back to it
+    @Test
+    public void shouldOfferBrokerApisThatAreUnknownToKroxy() {
+        try (var tester = mockKafkaKroxyliciousTester(KroxyliciousConfigUtils::proxy);
+                var client = tester.singleRequestClient()) {
+            ApiVersionsResponseData mockResponse = new ApiVersionsResponseData();
+            ApiVersionsResponseData.ApiVersion version = new ApiVersionsResponseData.ApiVersion();
+            version.setApiKey((short) 9999).setMinVersion((short) 3).setMaxVersion((short) 4);
+            mockResponse.apiKeys().add(version);
+            tester.setMockResponse(new Response(ApiKeys.API_VERSIONS, (short) 3, mockResponse));
+            Response response = whenGetApiVersionsFromKroxylicious(client);
+            assertEquals(ApiKeys.API_VERSIONS, response.apiKeys());
+            assertEquals((short) 3, response.apiVersion());
+            ApiVersionsResponseData message = (ApiVersionsResponseData) response.message();
+            assertEquals(1, message.apiKeys().size());
+            ApiVersionsResponseData.ApiVersion singletonVersion = message.apiKeys().iterator().next();
+            assertEquals((short) 9999, singletonVersion.apiKey());
+            assertEquals((short) 3, singletonVersion.minVersion());
+            assertEquals((short) 4, singletonVersion.maxVersion());
+        }
+    }
+
+    private static void givenMockRespondsWithApiVersionsForApiKey(MockServerKroxyliciousTester tester, ApiKeys keys, short minVersion, short maxVersion) {
+        ApiVersionsResponseData mockResponse = new ApiVersionsResponseData();
+        ApiVersionsResponseData.ApiVersion version = new ApiVersionsResponseData.ApiVersion();
+        version.setApiKey(keys.id).setMinVersion(minVersion).setMaxVersion(maxVersion);
+        mockResponse.apiKeys().add(version);
+        tester.setMockResponse(new Response(ApiKeys.API_VERSIONS, (short) 3, mockResponse));
```

Comment:
avoiding the magic number
```suggestion
        tester.setMockResponse(new Response(ApiKeys.API_VERSIONS, org.apache.kafka.common.message.ApiVersionsRequestData.HIGHEST_SUPPORTED_VERSION, mockResponse));
```